scripts/build.ps1

```diff
@@ -14,40 +14,49 @@ Write-Host "SkipTests=$SkipTests."
 
 if (!(Test-Path "dotnetcli\dotnet.exe")) {
     Write-Host "dotnet.exe not installed, downloading and installing."
-    if ($Version -eq "<default>") {
-        $Version = (Get-Content "$PSScriptRoot\..\DotnetCLIVersion.txt" -Raw).Trim()
-    }
-    Invoke-Expression -Command "$PSScriptRoot\install-dotnet.ps1 -Channel master -Version $Version -InstallDir $PSScriptRoot\..\dotnetcli"
+    Invoke-Expression -Command "$PSScriptRoot\install-dotnet.ps1 -Version 1.0.0 -InstallDir $PSScriptRoot\..\dotnetcli"
     if ($lastexitcode -ne $null -and $lastexitcode -ne 0) {
-        Write-Error "Failed to install latest dotnet.exe, exit code [$lastexitcode], aborting build."
+        Write-Error "Failed to install framework version 1.0.0, exit code [$lastexitcode], aborting build."
         exit -1
     }
 
-    # Temporary workaround until CLI, Core-Setup, CoreFx are all in sync with the shared runtime.
-    $SharedVersion = (Get-Content "$PSScriptRoot\..\SharedRuntimeVersion.txt" -Raw).Trim()
-    Invoke-Expression -Command "$PSScriptRoot\install-dotnet.ps1 -Channel master -Version $SharedVersion -InstallDir $PSScriptRoot\..\dotnetcli -SharedRuntime"
+    Invoke-Expression -Command "$PSScriptRoot\install-dotnet.ps1 -Version 2.0.0 -InstallDir $PSScriptRoot\..\dotnetcli"
     if ($lastexitcode -ne $null -and $lastexitcode -ne 0) {
-        Write-Error "Failed to install latest 2.1.0 shared runtime (version $SharedVersion), exit code [$lastexitcode], aborting build."
+        Write-Error "Failed to install framework version 2.0.0, exit code [$lastexitcode], aborting build."
         exit -1
     }
+}
 
-    Invoke-Expression -Command "$PSScriptRoot\install-dotnet.ps1 -Version 1.0.0 -InstallDir $PSScriptRoot\..\dotnetcli"
+$dotnetExePath=".\dotnetcli\dotnet.exe"
+
+$cliVersion = Invoke-Expression "$dotnetExePath --version"
+if ($Version -eq "<default>") {
+    $Version = (Get-Content ".\DotnetCLIVersion.txt" -Raw).Trim()
+}
+if ($cliVersion -ne $Version) {
+    Write-Host "Newest version of dotnet cli not installed, downloading and installing."
+    Invoke-Expression -Command "$PSScriptRoot\install-dotnet.ps1 -Channel master -Version $cliVersion -InstallDir $PSScriptRoot\..\dotnetcli"
     if ($lastexitcode -ne $null -and $lastexitcode -ne 0) {
-        Write-Error "Failed to install framework version 1.0.0, exit code [$lastexitcode], aborting build."
+        Write-Error "Failed to install latest dotnet.exe, exit code [$lastexitcode], aborting build."
         exit -1
     }
+}
 
-    Invoke-Expression -Command "$PSScriptRoot\install-dotnet.ps1 -Version 2.0.0 -InstallDir $PSScriptRoot\..\dotnetcli"
+# Temporary workaround until CLI, Core-Setup, CoreFx are all in sync with the shared runtime.
+$installedRuntimeVersions = Invoke-Expression "$dotnetExePath --list-runtimes" | %{ $_.split(" ")[1] } | Select-Object
```

Comment:
What happens if $_.split(" ") results in an empty array? Should we guard against that?

Reply:
Powershell won't throw an error. We should be fine. Try `"".split(" ")[1]` in PowerShell.

Reply:
Yes, but `$installedRuntimeVersions.Contains($SharedVersion)` would throw if it was set to null.

Reply:
`$installedRuntimeVersions` can only be null if `Invoke-Expression "$dotnetExePath --list-runtimes"` didn't return any runtimes. Since we are installing the latest first, I wouldn't think that would be possible.

Reply:
You are right that in normal circumstances, we will have at least one runtime installed. But if we have only one runtime installed, wouldn't splitting on space and indexing at [1] still return null since we only have one element? Are there always at least 2 elements (some header text maybe)?

**As an aside:** What happens if `--list-runtimes` doesn't exist (for instance if 2.0 cli was installed)? What type of error will the user get?

Reply:
As long as there is one line with one space (even if there are no other characters surrounding it), there is no issue.

`--list-runtimes` returns an error message in 2.0 with a space, so there's no issue there either.

The only way this fails is if `--list-runtimes` returns absolutely nothing.

Reply:
If you want the check, I'll add the check. I just disagree that it's needed. :)

Reply:
I think we need to look through the entire list and find all the versions of Microsoft.NETCore.App.

Just looking at the the first one listed could get the wrong result.

`dotnet --list-runtimes`
```text
Microsoft.AspNetCore.All 2.1.0-preview2-final [D:\GitHub\Fork\corefxlab\dotnetcli\shared\Microsoft.AspNetCore.All]
Microsoft.AspNetCore.App 2.1.0-preview2-final [D:\GitHub\Fork\corefxlab\dotnetcli\shared\Microsoft.AspNetCore.App]
Microsoft.NETCore.App 1.0.4 [D:\GitHub\Fork\corefxlab\dotnetcli\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 1.1.1 [D:\GitHub\Fork\corefxlab\dotnetcli\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 2.0.0 [D:\GitHub\Fork\corefxlab\dotnetcli\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 2.1.0-preview2-26406-04 [D:\GitHub\Fork\corefxlab\dotnetcli\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 2.1.0-preview3-26416-01 [D:\GitHub\Fork\corefxlab\dotnetcli\shared\Microsoft.NETCore.App
```

Right now it is checking `2.1.0-preview2-final == 2.1.0-preview3-26416-01` and returning false, causing us to re-install the existing version.

Reply:
> If you want the check, I'll add the check. I just disagree that it's needed. :)

You are right, it is not necessary, as long as you have at least one version installed.